packages/client/src/Dashboard.tsx

```diff
@@ -115,18 +116,27 @@ export default function Dashboard({ accountId, navigate }: Props) {
   const conversations = data && data.account && data.account.conversations
   const [selected, dispatch] = Sel.useSelectedConversations(conversations)
 
-  let isStarred = false
+  let isStarred =
+    !!conversations &&
+    conversations
+      .filter(conversation =>
+        selected.some(conversationId => conversation.id === conversationId)
+      )
+      .reduce(
+        (isStarred, conversation) => isStarred || conversation.isStarred,
+        false
+      )
 
-  selected.forEach(select => {
-    conversations &&
-      conversations.forEach(conversation => {
-        if (conversation.id === select) {
-          if (conversation.isStarred) {
-            isStarred = true
-          }
-        }
-      })
-  })
+  //if not all selected star, we want to star instead of unstar
+  if (isStarred && selected.length > 1) {
+    isStarred =
```

Comment:
Assigning `isStarred` twice is a [code smell](https://martinfowler.com/bliki/CodeSmell.html). The logic for this second assignment is correct - using `every` instead of `reduce` covers the cases that you need. The first assignment is redundant. You can delete the first assignment, and delare `isStarred` with `const`.